src/m365/commands/request.ts

```diff
@@ -0,0 +1,187 @@
+      unknownOptionsNames.forEach(o => {
+        properties[o] = typeof unknownOptions[o] !== 'undefined';
+      });
+
+      Object.assign(this.telemetryProperties, properties);
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --url <url>'
+      },
+      {
+        option: '-m, --method [method]',
+        autocomplete: this.allowedMethods
+      },
+      {
+        option: '-r, --resource [resource]'
+      },
+      {
+        option: '-b, --body [body]'      
+      },
+      {
+        option: '-p, --filePath [filePath]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const currentMethod = args.options.method || 'get';
+        if (this.allowedMethods.indexOf(currentMethod) === -1) {
+          return `${currentMethod} is not a valid value for method. Allowed values: ${this.allowedMethods.join(', ')}`;
+        }
+
+        if (args.options.body && (!args.options.method || args.options.method === 'get')) {
+          return 'Specify a different method when using body';
+        }
+
+        if (args.options.body && !args.options['content-type']) {
+          return 'Specify the content-type when using body';
+        }
+
+        if (args.options.filePath && !fs.existsSync(path.dirname(args.options.filePath))) {
+          return 'Specified filePath to save the file does not exist';
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
+    if (this.verbose) {
+      logger.logToStderr(`Preparing request...`);
+    }
+
+    const method = (args.options.method || 'get').toUpperCase();
+    const headers: AxiosRequestHeaders = {};
+
+    const unknownOptions: any = this.getUnknownOptions(args.options);
+    const unknownOptionsNames: string[] = Object.getOwnPropertyNames(unknownOptions);
+    unknownOptionsNames.forEach(o => {
+      headers[o] = unknownOptions[o];
+    });
+
+    if (!headers.accept) {
+      headers.accept = 'application/json';
+    }
+
+    if (args.options.resource) {
+      headers['x-resource'] = args.options.resource;
+    }
+
+    const config: AxiosRequestConfig<string> = {
+      url: args.options.url,
+      headers,
+      method,
+      data: args.options.body
+    };
+
+    if (headers.accept.toString().startsWith('application/json')) {
+      config.responseType = 'json';
+    }
+
+    if (args.options.filePath) {
+      config.responseType = 'stream';
+    }
+
+    if (this.verbose) {
+      logger.logToStderr(`Executing request...`);
+    }
+
+    if (args.options.filePath) {
+      request
+        .execute<any>(config)
+        .then((file: any): Promise<string> => {
+          return new Promise((resolve, reject) => {
+            const writer = fs.createWriteStream(args.options.filePath as string);
+
+            file.data.pipe(writer);
```

Comment:
This isn't giving the correct output. When I call:

```sh
m365 request --url "https://graph.microsoft.com/beta/me/photo/$value" --filePath ./picture.jpg
```

The picture.jpg file stored on disk contains:

```json
{"@odata.context":"https://graph.microsoft.com/beta/$metadata#users('e2154125-9f63-4cc2-a7b4-f4f8adadea81')/photo/$entity","@odata.mediaContentType":"image/jpeg","@odata.mediaEtag":"W/\"3704be3eec40881fecd9c508b6905c2079154e42eacf3dae5e719e275a83c554\"","id":"default","height":279,"width":280}
```

Not the actual binary contents of the image. The request is giving correct response, but we need to ensure that we only store the payload and not the metadata.

Reply:
That's because you need to escape the dollar sign...

Reply:
This proves we need an extra example.

Reply:
Ah right, totally didn't occur to me. Yes, we should definitely have a sample around this, because it will bite us guaranteed.